eth_tester/backends/pyevm/main.py

```diff
@@ -419,18 +427,20 @@ def _max_available_gas(self):
         return header.gas_limit - header.gas_used
 
     def estimate_gas(self, transaction):
-        # TODO: move this to the VM level (and use binary search approach)
-        signed_evm_transaction = self._get_normalized_and_signed_evm_transaction(
+        evm_transaction = self._get_normalized_and_unsigned_evm_transaction(
             dict(transaction, gas=self._max_available_gas()),
         )
 
-        computation = _execute_and_revert_transaction(self.chain, signed_evm_transaction, 'pending')
-        if computation.is_error:
-            raise TransactionFailed(str(computation._error))
-
-        gas_used = computation.gas_meter.start_gas - computation.gas_meter.gas_remaining
+        spoof_evm_transaction = SpoofTransaction(
+            evm_transaction,
+            get_sender=lambda: transaction['from'],
+            sender=transaction['from'],
+            intrinsic_gas=21000,
```

Comment:
This probably shouldn't be hard-coded, in case future hard forks change it. Can't we just leave it as the default from whatever `evm_transaction` has?